### ISSUE TITLE
Update Cogs for New d.py Changes

### ISSUE THREAD

**Issue by sco1** [python-discord/bot#319]
d.py rewrite introduced some breaking changes to cogs in [caf3d17](https://github.com/Rapptz/discord.py/commit/caf3d17d4aa3ce45435bb96ea6d99317bfef8618) that need to be mitigated before we can bump our locked version of the library.

A [migration guide](https://discordpy.readthedocs.io/en/rewrite/migrating.html#cog-changes) can be found in the documentation.

**Comment by SebastiaanZ** [comment]
Sound like quite a project. We should probably list all the things we need to change, so we can check them off. I'll put in an issue for the next staff meeting as well. I feel pretty useless for the django move at the moment, so maybe some of us can already start looking into this.

**Comment by mathsman5133** [comment]
I am happy to tackle this and migrate all bot code to v1.2.3 (most recent) version of d.py, if @SebastiaanZ hasn't started.

**Comment by lemonsaurus** [comment]
I would welcome this, and you'd earn yourself a Contributor role in so doing, @mathsman5133

**Comment by SebastiaanZ** [comment]
This should probably wait until the `django` branch and the `master` branch have merged, though. It's going to create massive merge conflicts otherwise.

**Comment by lemonsaurus** [comment]
Not sure about massive merge conflicts but I do agree. Where are we on that? @MarkKoz?

**Comment by mathsman5133** [comment]
Sounds good 👍 Is there a rough timeline for django/master push?

**Comment by MarkKoz** [comment]
I haven't finished the merge since it's waiting on nomination stuff. I could just leave that one cog unmerged and push the rest of the merged changes if you think that's best.

**Comment by lemonsaurus** [comment]
@MarkKoz @SebastiaanZ can we expedite the nomination stuff? Anything I can do to assist?

**Comment by SebastiaanZ** [comment]
It's mainly hanging on the back-end API PR, I think, and I have no idea in which direction we want to take that. The bot part looks okay, but obviously depends on the API endpoints, so that needs to be figured out and coded first. 

In addition, I won't have access to a machine with a Python environment for most of this week and the upcoming weekend.

**Comment by lemonsaurus** [comment]
hm. that's too bad. okay in that case, I think we should do it as you suggested, @MarkKoz. Get everything except nomination stuff pushed so that @mathsman5133 can get to work porting to d.py 1.2.3.

**Comment by lemonsaurus** [comment]
as for what direction to take it, @SebastiaanZ, I really don't think it matters that much so long as it works. From what I gather the difference between the one direction and the other are largely semantic and not really major enough for us to spend any more time on. You should just head in the direction of least friction and get it merged, imo.

**Comment by SebastiaanZ** [comment]
I've pushed three versions that were all fully functional; I'm really not sure in which direction I should go next.

**Comment by lemonsaurus** [comment]
Then I suggest we go with whatever is the least work, so let's just merge what you've got now if it's fully functional, in the interest of getting stuff moving along. Our eyes should be on the goal now that we're so close to feature unfreeze.

**Comment by mathsman5133** [comment]
I can start plugging away at this over the next few weeks, and I'm correct in saying I'm working from the `django` branch?
 
Also - do you want the whole lot in 1 commit, 1 commit per-file, 1 commit per-version change (ie. one for changes to migrate to v1.0.0, another for v1.1.0 etc.) or some other format?

Lastly, I get a funny error when running the bot. Somewhere on discord may be more appropriate but I'll chuck it here anyway in case it's something super simple.
When running `pipenv run start`:
```
aiohttp.client_exceptions.ClientConnectorSSLError: Cannot connect to host api.pythondiscord.local:8080 ssl:None [[SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:841)]
```
The site is running fine (with `pipenv run rundev`) and I can get around everything at `pythondiscord.local:8080`.
When trying to get to `http://api.pythondiscord.local:8080/`, however, I get this:
```
{
  "error_code": 0, 
  "error_message": "Unknown API route"
}
```
Windows Pro x64 / Pycharm 2019 w/ all relevant pipfile deps

**Comment by MarkKoz** [comment]
> I can start plugging away at this over the next few weeks, and I'm correct in saying I'm working from the django branch?

Yes, target the `django` branch.

> Also - do you want the whole lot in 1 commit, 1 commit per-file, 1 commit per-version change (ie. one for changes to migrate to v1.0.0, another for v1.1.0 etc.) or some other format?

We're not strict on this but 1 commit per file sounds OK. 1 commit per-version change sounds like it'd be too much of a hassle for you.

> Lastly, I get a funny error when running the bot.

Did you change the config to use `http` instead of `https`?

**Comment by lemonsaurus** [comment]
@mathsman5133 FYI, after some internal discussion, we've decided that we won't merge any d.py migration changes until we've completed the move to Django. The `django` branch is already quite large, and merging in d.py migration changes will make it harder to review. 

This shouldn't affect your ability to work on it, but it might mean that the PR will have to stay open a little longer than it normally would if you finish it before we're out of feature freeze.
